lib/channels/app_socket.ex

```diff
@@ -43,12 +40,17 @@ defmodule ApplicationRunner.AppSocket do
       # See `Phoenix.Token` documentation for examples in
       # performing token verification on connect.
       @impl true
-      def connect(params, socket, _connect_info) do
-        with {:ok, app_name, context} <- extract_params(params),
-             {:ok, user_id} <- @adapter_mod.resource_from_params(params),
-             :ok <- @adapter_mod.allow(user_id, app_name),
+      def connect(adapter_mod, params, socket, _connect_info) do
+        with {:ok, app_name, context} <- ApplicationRunner.AppSocket.extract_params(params),
+             {:ok, user_id} <- adapter_mod.resource_from_params(params),
+             :ok <- adapter_mod.allow(user_id, app_name),
              {:ok, env_metadata, session_metadata} <-
-               create_metadatas(user_id, app_name, context),
+               ApplicationRunner.AppSocket.create_metadatas(
```

Comment:
There is an alias for AppSocket so you can use it (not very important):

```suggestion
        with {:ok, app_name, context} <- AppSocket.extract_params(params),
             {:ok, user_id} <- adapter_mod.resource_from_params(params),
             :ok <- adapter_mod.allow(user_id, app_name),
             {:ok, env_metadata, session_metadata} <-
               AppSocket.create_metadatas(
```